pyhanko/cli.py

```diff
@@ -447,6 +447,11 @@ def validate_signatures(ctx, infile, executive_summary,
                         soft_revocation_check, no_revocation_check, password,
                         retroactive_revinfo, detached):
 
+    def exit_if_invalid(signature_ok):
```

Comment:
A couple things:

 - According to a comment in `sysexits.h` on my system, `EX_DATAERR` is for data format errors. Most signature validation errors are not of this type. Also, in pyHanko's current validation model (which is a little primitive), signatures may be judged as invalid due to environmental factors preventing validation from being completed (e.g. lack of available revocation information, or missing intermediate certificates). Using `EX_DATAERR` for that seems particularly inappropriate to me.
 - I'm not sure how I feel about `exit_if_invalid` being a function. Things that look like functions shouldn't silently kill the program---see comment below for a suggested alternative.

Reply:
Thanks! Agreed about the not going for anything stranger than 1 as exit code, and removing the `exit_if_invalid` in favour of an exception.